### PR TITLE
Meson Windows build and CI

### PR DESCRIPTION
**Pull request by matthew-brett**
* Set long double to double with compile flags.
* Relax threshold for hyp2f1 slightly.
* Disable pocketfft threading for now, for mingw build.
* Add Github actions CI for Windows.

### CODE REVIEW

**Comment by matthew-brett** [comment]
I believe the macOS tests are unrelated - at least, I couldn't see any of these changes that could have caused the macOS errors.

File: .github/workflows/windows.yml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+          gdown --id 1G3jwxMY4ZVRqvz6owl0kmw_Innmr4ZnV
+          unzip -d c:\ openblas_32_if.zip
+          echo "PKG_CONFIG_PATH=c:\opt\openblas\if_32\64\lib\pkgconfig;" >> $env:GITHUB_ENV
+      - name: meson-configure
+        run: |
+          git submodule update --init --recursive
+          meson build --prefix=$PWD\build
+      - name: meson-build
+        run: |
+          ninja -j 128 -C build
```
Comment (rgommers) [inline .github/workflows/windows.yml:51]: What's the reason for `-j 128`? Isn't that going to be quite slow compared to `-j 2`? Maybe a left-over test setting?

File: .github/workflows/windows.yml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+          gcc --version
+          gfortran --version
+      - name: pip-packages
+        run: |
+          pip install numpy cython pybind11 pythran meson ninja
+      - name: openblas-libs
+        run: |
+          choco install unzip
+          pip install gdown
+          gdown --id 1G3jwxMY4ZVRqvz6owl0kmw_Innmr4ZnV
```
Comment (rgommers) [inline .github/workflows/windows.yml:42]: This could use a comment about this download containing only openblas and how it was built.

**Review by rgommers — APPROVED** [review]
This is great, thanks @matthew-brett (and @carlkl as well!). I added 2 comments that would be nice to address, but they're not blocking. CI is not running on a PR, so let's merge this and see how it does.

File: .github/workflows/windows.yml (hunk 1 above)
Comment (matthew-brett) [reply .github/workflows/windows.yml:51]: I tried -j 64 but still got the Cython out-of-order build errors...

File: .github/workflows/windows.yml (hunk 2 above)
Comment (matthew-brett) [reply .github/workflows/windows.yml:42]: Yes - I'll do one - thanks.  I'll also suggest this build procedure over at openblas-libs.

File: .github/workflows/windows.yml (hunk 1 above)
Comment (rgommers) [reply .github/workflows/windows.yml:51]: Ah makes sense, you do all jobs in parallel and then the fastest one gets triggered in time. Decent workaround.

**Comment by rgommers** [comment]
I pushed a small fix to the CI config (branch names to trigger on), and now the job passes: https://github.com/rgommers/scipy/actions/runs/1542031258

**Comment by matthew-brett** [comment]
Nice - thanks.

**Pull request merged**